hil/ext/switches/dellnos9.py

```diff
@@ -61,6 +62,24 @@ def validate(kwargs):
     def session(self):
         return self
 
+    def ensure_legal_operation(self, nic, op_type, channel):
+        table = model.NetworkAttachment
+        if channel != 'vlan/native' and op_type == 'connect' and \
+            db.session.query(table).filter(table.nic_id == nic.id) \
+                .filter(table.channel == 'vlan/native').count() == 0:
```

Comment:
this overly complex db query is performed twice. make a private helper method with a descriptive and meaningful name.

Reply:
the queries are slightly different though. One checks for `channel = 'vlan/native'` while the other one checks for the opposite. The only thing I could put in a helper method is the first part of the query where it finds the `nic_id` in table. Should I do that then?